server/middlewares/clientAuth.ts

```diff
@@ -0,0 +1,18 @@
+import { Response, NextFunction } from 'express';
+
+import { AuthRequestInterface } from '../utils/interfaces/authRequestInterface';
+import { CustomError } from '../utils/custom_error';
+
+export const clientAuthController = (req: AuthRequestInterface, res: Response, next: NextFunction): void => {
+  try {
+    const { user } = req;
+
+    if (user?.role === 'client') {
```

Comment:
if its a role system and you're gonna have multiple roles, it doesnt make sense to create one for each one of the roles
create a function that returns a middleware!
```ts
const isRole = (role) => (req,res,next) => {
...
if (user?.role === role) {
...
}
```

Reply:
I'm using these middlewares in the route middleware chain to not let the request reach the main middleware if the user is not authorized, if I make it this way, I'm not going to be able to use it like that because how am I gonna pass invoke the function and pass the role to it.

Reply:
you can use it then later like: `router.post('get-user', isRole("admin"), finalcontroller)`

Reply:
Got it.